iothub_client/tests/common_longhaul/iothub_client_common_longhaul.c

```diff
@@ -399,14 +401,48 @@ static char* create_twin_desired_properties_update(const char* test_id, unsigned
     return result;
 }
 
+static void validate_internet_conductivity()
+{
+    HTTPAPIEX_HANDLE httpExApiHandle;
+    unsigned int statusCode = 0;
+    BUFFER_HANDLE responseBuffer;
+
+    if ((responseBuffer = BUFFER_new()) == NULL)
+    {
+        LogError("BUFFER_new failed for responseBuffer");
+    }
+    else if ((httpExApiHandle = HTTPAPIEX_Create("bing.com")) == NULL)
+    {
+        LogError("HTTPAPIEX_Create failed");
+    }
+    else if (HTTPAPIEX_ExecuteRequest(httpExApiHandle, HTTPAPI_REQUEST_GET, "", NULL, NULL, &statusCode, NULL, responseBuffer) != HTTPAPIEX_OK)
+    {
+        LogError("validate_internet_conductivity: bing.com is NOT accessible!");
+    }
+    else
+    {
+        BUFFER_delete(responseBuffer);
+        if (statusCode >= 200 && statusCode < 300)
+        {
+            LogInfo("validate_internet_conductivity: bing.com is accessible");
+        }
+    }
+}
+
 static void connection_status_callback(IOTHUB_CLIENT_CONNECTION_STATUS status, IOTHUB_CLIENT_CONNECTION_STATUS_REASON reason, void* userContextCallback)
 {
-    IOTHUB_LONGHAUL_RESOURCES* iotHubLonghaul = (IOTHUB_LONGHAUL_RESOURCES*)userContextCallback;
+    LogInfo("Device Connection Status: status=%s, reason=%s", MU_ENUM_TO_STRING(IOTHUB_CLIENT_CONNECTION_STATUS, status), MU_ENUM_TO_STRING(IOTHUB_CLIENT_CONNECTION_STATUS_REASON, reason));
 
+    IOTHUB_LONGHAUL_RESOURCES* iotHubLonghaul = (IOTHUB_LONGHAUL_RESOURCES*)userContextCallback;
     if (iothub_client_statistics_add_connection_status(iotHubLonghaul->iotHubClientStats, status, reason) != 0)
     {
-        LogError("Failed adding connection status statistics (%s, %s)",
-            MU_ENUM_TO_STRING(IOTHUB_CLIENT_CONNECTION_STATUS, status), MU_ENUM_TO_STRING(IOTHUB_CLIENT_CONNECTION_STATUS_REASON, reason));
+        LogError("Failed adding connection status statistics");
+    }
+
+    if (reason == IOTHUB_CLIENT_CONNECTION_COMMUNICATION_ERROR ||
+        reason == IOTHUB_CLIENT_CONNECTION_NO_NETWORK)
+    {
+        validate_internet_conductivity();
```

Comment:
should the test fail if this fails? aka should this function return a value and take action?

Reply:
not right now. I want to validate if the host can access the internet or is this just a IoT service issue. Currently used to validate the logs.